### ISSUE TITLE
Open multiple files with one action

### ISSUE THREAD

**Issue by ben-burlingham** [atom/tree-view#879]
https://discuss.atom.io/t/open-multiple-files-in-one-action-from-the-tree-view/30479

Opening the contents of an entire folder is a use case I run into fairly often.

Currently, the files must be opened individually, which takes some patience.

There is already `Shift` select and a `Cmd` select handling of multiple files, so it seems that not being able to open the selection in one go is counter-intuitive.

Thanks everyone for your contributions to this great editor!

-----

**Repro Steps**

1. Ensure that `core.allowPendingPaneItems` is *enabled*
1. Navigate in the terminal to your local atom repository
1. Launch Atom using `atom .`
1. In the Tree View, <kbd>Cmd+Click</kbd> the files `CHANGELOG.md`, `CODE_OF_CONDUCT.md` and `CONTRIBUTING.md`
1. Press <kbd>Enter</kbd>

**Expected:** All three files to open

**Actual:** Initially selected file is confirmed, no additional files are opened

**Comment by 50Wliu** [comment]
This is a duplicate of #575 - feel free to subscribe there for updates.

**Comment by ben-burlingham** [comment]
@50Wliu it looks like https://github.com/atom/tree-view/issues/575 is titled "Select multiple files with shift up/down", which is a different issue.

The description in that ticket describes holding shift and using the arrows to select.

My issue is that once they are selected, multiple files can't be opened.

Are you sure this is a duplicate?

**Comment by lee-dohm** [comment]
@ben-burlingham You are correct, this isn't a duplicate. I've added steps to reproduce the desired behavior. This is a tool that makes even enhancement requests much easier for people to understand :grinning:

**Comment by Ben3eeE** [comment]
The pending pane items setting does not change the behavior for me, it is always the initially selected file that opens regardless if `core.allowPendingPaneItems` is _enabled_ or _disabled_.

**Comment by Yopai** [comment]
I agree, this feature would be a lot interesting !
As it was said, it is really counter-intuitive and rather annoying to have to double-click each file (if I simple click, file is open only temporarily, with name in italics)

**Comment by Ben3eeE** [comment]
> (if I simple click, file is open only temporarily, with name in italics)

You can change this behavior so the file opens with single click with the [single-click-open](https://atom.io/packages/single-click-open) package.

**Comment by ben-burlingham** [comment]
Thanks @Ben3eeE, that cuts out half of the clicking work, at least

**Comment by AndreaGriffiths11** [comment]
I was able to reproduce this using Atom v1.13.1 on Mac OS X 10.12.3.

**Comment by neurogenesis** [comment]
Would it be possible to just merge something like the following into core Atom?

https://github.com/hikaruworld/atom-multi-file-open

**Comment by jpickwell** [comment]
Atom: v1.36.0

Still not able to open multiple selected files. I've tried pressing Enter key, and dragging selected files to the editor pane. Enter key only opens the first, initial file selected. Drag-n-drop does nothing.

**Comment by DilumAluthge** [comment]
+1

Is there any plan to implement this feature?
